src/promptflow-devkit/promptflow/_cli/_pf/_flow.py

```diff
@@ -510,18 +514,20 @@ def _test_flow_multi_modal(args, pf_client):
         from promptflow._sdk._tracing import _invoke_pf_svc
 
         # Todo: use base64 encode for now, will consider whether need use encryption or use db to store flow path info
-        def generate_url(flow_path, port):
+        def generate_url(flow_path, port, url_params):
             encrypted_flow_path = encrypt_flow_path(flow_path)
             query_dict = {"flow": encrypted_flow_path}
             if Configuration.get_instance().is_internal_features_enabled():
                 query_dict.update({"enable_internal_features": "true"})
             query_params = urlencode(query_dict)
+            if url_params:
```

Comment:
maybe url-params can be a dict and use urlencode to generate query?

Reply:
this won't impact user interface (how UX pass in the param). let me do this in next PR.